my-ansible-playbook.yml

```diff
@@ -0,0 +1,58 @@
+- name: Install To-Do App on new web servers
+  hosts: webservers
+  remote_user: ec2-user
+  vars_prompt:
+    - name: trello_key  
+      prompt: Enter the trello key
+      private: true
+    - name: trello_token
+      prompt: Enter the trello token
+      private: true
+
+  tasks:
+    - name: Install packages
+      ansible.builtin.yum:
+        name: "{{ packages }}"
+      vars:
+        packages:
+        - git
+        - python3
+      become: yes
+    - name: Install poetry
+      ansible.builtin.shell:
+        cmd: curl -sSL https://install.python-poetry.org | python3 -
+    - name: Create directory for the To-Do App code
+      ansible.builtin.file:
+        path: /opt/todoapp
+        state: directory
+      become: yes
+    - name: Get the latest version of your code with Git
+      ansible.builtin.git:
+        repo: 'https://github.com/shaylipatel/DevOps-Course-Starter.git'
+        dest: /opt/todoapp
+        version: exercise-4
+      become: yes
+    - name: Install project dependencies
+      ansible.builtin.command:
+        chdir: /opt/todoapp
+        cmd: /home/ec2-user/.local/bin/poetry install
+      become: yes
```

Comment:
From past experience, I expect a few of these commands should be able to run without `become: yes`, but it may require tweaking some of the folder permissions (e.g. setting the owner for the `todoapp` folder as below). I think not necessarily worth actively changing, but worth bearing in mind, as in a real project having everything run as the sudo user is a bad security habit

```yml
      path: /opt/todoapp
      state: directory
      owner: ec2-user
      mode: 0755
```

Reply:
Noted, amended owner of file and permission and have been able to remove some of the `become: yes`